core/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/cancel/TransportCancelTasksAction.java

```diff
@@ -322,5 +316,15 @@ public void messageReceived(final BanParentTaskRequest request, final TransportC
         }
     }
 
+    /**
+     * Returns a set of all currently available nodes in the cluster
+     */
+    private Set<String> nodes() {
+        Set<String> nodes = new HashSet<>();
```

Comment:
It feels weird to turn the nodes list in the cluster state in set of strings and then turn it back into a node on `sendSetBanRequest`.